src/main/java/org/entando/plugins/pda/pam/service/group/KieGroupService.java

```diff
@@ -0,0 +1,34 @@
+package org.entando.plugins.pda.pam.service.group;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.entando.plugins.pda.core.engine.Connection;
+import org.entando.plugins.pda.core.service.group.GroupService;
+import org.entando.plugins.pda.pam.service.api.CustomQueryService;
+import org.entando.plugins.pda.pam.service.api.KieApiService;
+import org.kie.server.api.model.definition.AssociatedEntitiesDefinition;
+import org.kie.server.client.ProcessServicesClient;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class KieGroupService implements GroupService {
+
+    private final CustomQueryService customQueryService;
+    private final KieApiService kieApiService;
+
+    @Override
+    public List<String> list(Connection connection, String containerId, String processId) {
+        if (containerId != null && processId != null) {
```

Comment:
Should we permit listing groups from all processes inside a container or all processes with id from all containers?

Reply:
Right now I don't think so. It will complicate the implementation, make many calls to the kie server and we don't even have a use case for it.